service/backend/pchain/account.go

```diff
@@ -162,6 +169,98 @@ func (b *Backend) AccountCoins(ctx context.Context, req *types.AccountCoinsReque
 	}, nil
 }
 
+func (b *Backend) getPendingRewardsBalance(ctx context.Context, req *types.AccountBalanceRequest) (*types.AccountBalanceResponse, *types.Error) {
+	addr, err := address.ParseToID(req.AccountIdentifier.Address)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInvalidInput, "malformed address")
+	}
+
+	metadata := req.AccountIdentifier.Metadata
+	var validatorNodeID ids.NodeID
+	var nodeIDs []ids.NodeID
+	if metadata != nil && metadata[pmapper.MetadataValidatorNodeID] != nil {
+		nodeIDStr, ok := metadata[pmapper.MetadataValidatorNodeID].(string)
+		if !ok {
+			return nil, service.WrapError(service.ErrInvalidInput, "validator_node_id must be a string")
+		}
+
+		validatorNodeID, err = ids.NodeIDFromString(nodeIDStr)
+		if err != nil {
+			return nil, service.WrapError(service.ErrInvalidInput, "malformed validator_node_id ")
+		}
+
+		nodeIDs = append(nodeIDs, validatorNodeID)
+	}
+
+	validators, err := b.pClient.GetCurrentValidators(ctx, constants.PrimaryNetworkSubnetID, nodeIDs)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInternalError, "unable to fetch validators")
+	}
+
+	validatorRewards := big.NewInt(0)
+	delegationFeeRewards := big.NewInt(0)
+	delegationRewards := big.NewInt(0)
+
+	for _, v := range validators {
+		isValidatorOwner := false
+
+		if slices.Contains(v.ValidationRewardOwner.Addresses, addr) {
+			validatorRewards.Add(validatorRewards, new(big.Int).SetUint64(*v.PotentialReward))
+			isValidatorOwner = true
+		}
+
+		for _, d := range v.Delegators {
```

Comment:
@bilgehansahin due to recent changes to `GetCurrentValidators` API you won't get full delegators information if you specify multiple validators (or all validators with the empty parameter). I reckon this is annoying, but I think you need to query specific validators here, one by one.

Reply:
Interesting timing. Will make the NodeID a required parameter in this case. Was this due to node performance concerns?

Reply:
If you look at my last change, now user must specify a single validator node id specified as the `sub_account.address`.